src/core/ext/transport/chttp2/transport/hpack_encoder.cc

```diff
@@ -42,6 +42,12 @@
 #include "src/core/lib/transport/static_metadata.h"
 #include "src/core/lib/transport/timeout_encoding.h"
 
+/* don't consider adding anything bigger than this to the hpack table */
+#define MAX_DECODER_SPACE_USAGE 512
```

Comment:
nit: now that you're refactoring, these macros can be `constexpr` inside the anonymous namespace.